### PR TITLE
Make published artifacts full-fledged

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR addresses [#333](https://github.com/SpineEventEngine/config/issues/333).

In particular, this changeset does the following:

1. Pulls the latest `config` to obtain [the changes, made to our publishing](https://github.com/SpineEventEngine/config/pull/335).
2. Drops the dead code from `publish` package in `buildSrc`.
3. Slightly refactors root's build file. 
4. Explicitly declares which modules are published.
5. [Bumps versions](https://github.com/SpineEventEngine/config/pull/344) of some dependencies.

Please note that `proto.jar` disappeared from the list of artifacts in `plugin-base` and `plugin-testlib` modules. These modules don't contain any Proto files in theirs `main` source sets:

image

Previously, their `proto.jar`s contained Proto declarations from their dependencies. In particular, from `base` library. Which is kind of unexpected. This artifact was not meant as "uber-jar" for Proto files.

The library version is bumped to `2.0.0-SNAPSHOT.91`.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]
LGTM.

Also, as discussed, let's create an issue in our configuration repo describing the inability to address the detected Error Prone vulnerability until they release a new version.

@alexander-yevsyukov FYI on Error Prone.

File: pom.xml [hunk 1]
```diff
@@ -122,18 +122,18 @@ all modules and does not describe the project structure per-subproject.
   <dependency>
     <groupId>com.google.errorprone</groupId>
     <artifactId>error_prone_annotations</artifactId>
-    <version>2.10.0</version>
+    <version>2.11.0</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:128]: @sonatype-lift ignore

File: pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@ all modules and does not describe the project structure per-subproject.
...
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:26]: @sonatype-lift ignore

File: pom.xml [hunk 3]
```diff
@@ -26,7 +26,7 @@ all modules and does not describe the project structure per-subproject.
...
     <groupId>com.google.guava</groupId>
     <artifactId>guava</artifactId>
-    <version>31.0.1-jre</version>
+    <version>31.1-jre</version>
     <scope>compile</scope>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:32]: @sonatype-lift ignore

File: pom.xml [hunk 4]
```diff
@@ -122,29 +122,29 @@ all modules and does not describe the project structure per-subproject.
...
+    <version>2.11.0</version>
     <scope>provided</scope>
   </dependency>
   <dependency>
     <groupId>com.google.protobuf</groupId>
     <artifactId>protoc</artifactId>
-    <version>3.19.3</version>
+    <version>3.19.4</version>
   </dependency>
   <dependency>
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:144]: @sonatype-lift ignore

**Comment by yevhenii-nadtochii** [comment]
@armiol Please take another look.

I've suppressed already known warnings from LiftBot. 

The rest warning reports that the last version of `PMD` uses version of `commons-io` that has a vulnerability. In their `POM` file they state that the problematic version of `commons-io` is the last one that supports Java 7 ...

```
            <dependency>
                <groupId>commons-io</groupId>
                <artifactId>commons-io</artifactId>
                <version>2.6</version> <!-- note: this is the last version compatible with java7 -->
            </dependency>
```

**Review by armiol — APPROVED** [review]
@yevhenii-nadtochii LGTM.

Again, as long as we use the latest PMD (and we do; more than that, it's been recently released!), we aren't able to do much more. So, let's ignore.

File: pom.xml [hunk 5]
```diff
@@ -154,12 +154,12 @@ all modules and does not describe the project structure per-subproject.
...
```
[the remark this answers is not in the document]
Comment (yevhenii-nadtochii) [reply pom.xml:154]: @sonatype-lift ignore

**Pull request merged**